### PR TITLE
test(core): integrate Testcontainers and enhance seed data

### LINKED ISSUE #3: Review constraint checks and pontentially remove mocks
A placeholder issue to:
- [x] Review the constraint check algorithm
- [x] Check potential (depending on time) to remove mocks and replace with real test database tests
- [x] Review test cases and find other edge cases

### PR DESCRIPTION
**Pull request by nigelnindodev**
Closes #3 

## Summary by CodeRabbit

* **Chores**
  * Added containerized PostgreSQL support and test lifecycle hooks for reliable test runs.
  * Added a DB truncate helper used by tests.

* **Tests**
  * Converted many unit tests into DB-backed integration tests that seed and validate real entities.
  * Added global setup/teardown to manage the test database lifecycle and reset DB between tests.

* **Data Updates**
  * Expanded demo dataset with a new staff member and multi-location, multi-day shift scenarios.

* **Refactor**
  * Centralized scheduling constraint codes/types and standardized constraint results.

* **Documentation**
  * Added server-side agent guidelines and best practices.

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -98,6 +100,8 @@
       "**/*.(t|j)s"
     ],
     "coverageDirectory": "../coverage",
-    "testEnvironment": "node"
+    "testEnvironment": "node",
+    "globalSetup": "../test/global-setup.ts",
+    "globalTeardown": "../test/global-teardown.ts"
```
[the remark this answers is not in the document]
Comment (nigelnindodev) [reply server/package.json]: Yes, you can open an issue.
